### PR TITLE
Updated GeoVector with new functionality. Added tests for GeoVector.

### PR DESCRIPTION
**Pull request by RubyNova**

### CODE REVIEW

File: include/NovelRT/Maths/GeoVector.h [hunk 1]
```diff
@@ -59,6 +59,14 @@ namespace NovelRT::Maths {
       _value.y = value;
     }
 
+    inline GeoVector<T> getNormalised() const noexcept {
+      return GeoVector<T>(glm::normalize(getVec2Value()));
+    }
+
+    inline float getMagnitude() const noexcept {
```
Comment (tannergooding) [inline include/NovelRT/Maths/GeoVector.h]: nit: I think `length` is the more standard term

File: include/NovelRT/Maths/GeoVector.h [hunk 2]
```diff
@@ -99,9 +107,21 @@ namespace NovelRT::Maths {
...
-    void rotateToAngleAroundPoint(T angleRotationValue, const GeoVector<T>& point) {
+    void rotateToAngleAroundPoint(T angleRotationValue, const GeoVector<T>& point) noexcept {
       setVec2Value(glm::rotate((getVec2Value() - point.getVec2Value()), glm::radians(angleRotationValue)) + point.getVec2Value());
     }
+
+    bool epsilonEquals(const GeoVector<T>& other, const GeoVector<T>& epsilonValue) const noexcept {
+      return glm::all(glm::equal(getVec2Value(), other.getVec2Value(), epsilonValue.getVec2Value()));
+    }
+
+    static GeoVector<T> zero() {
```
Comment (tannergooding) [inline include/NovelRT/Maths/GeoVector.h]: why not have these as const?

Comment (RubyNova) [reply include/NovelRT/Maths/GeoVector.h]: statics can't have those sorts of qualifiers, according to the compiler.

File: include/NovelRT/Maths/GeoVector.h (hunk 1 above)
Comment (RubyNova) [reply include/NovelRT/Maths/GeoVector.h]: I've only ever heard it addressed as magnitude in a game dev context, but I can change it if you wish.

File: include/NovelRT/Maths/GeoVector.h (hunk 2 above)
Comment (tannergooding) [reply include/NovelRT/Maths/GeoVector.h]: no, I meant as a const field rather than a method

File: include/NovelRT/Maths/GeoVector.h (hunk 1 above)
Comment (tannergooding) [reply include/NovelRT/Maths/GeoVector.h]: both glm and dxmath use the term `length` as do `glsl` and `hlsl`

File: include/NovelRT/Maths/GeoVector.h (hunk 2 above)
Comment (RubyNova) [reply include/NovelRT/Maths/GeoVector.h]: As per the discussion in discord, this was causing compiler errors. Will be resolved at a later date.

File: include/NovelRT/Maths/GeoVector.h (hunk 1 above)
Comment (RubyNova) [reply include/NovelRT/Maths/GeoVector.h]: Added an alias for `getMagnitude`.

**Pull request merged**